src/app/src/main/kotlin/org/vaccineimpact/orderlyweb/db/repositories/TagRepository.kt

```diff
@@ -0,0 +1,53 @@
+package org.vaccineimpact.orderlyweb.db.repositories
+
+import org.vaccineimpact.orderlyweb.db.JooqContext
+import org.vaccineimpact.orderlyweb.db.Tables
+import org.vaccineimpact.orderlyweb.db.Tables.ORDERLYWEB_REPORT_TAG
+import org.vaccineimpact.orderlyweb.db.Tables.ORDERLYWEB_REPORT_VERSION_TAG
+
+interface TagRepository
+{
+    fun getReportTags(reportNames: List<String>): Map<String, List<String>>
+    fun tagReport(reportName: String, tag: String)
+    fun tagVersion(versionId: String, tag: String)
+}
+
+class OrderlyWebTagRepository : TagRepository
+{
+    override fun getReportTags(reportNames: List<String>): Map<String, List<String>>
+    {
+        JooqContext().use { ctx ->
+            return ctx.dsl.select(
+                    Tables.ORDERLYWEB_REPORT_TAG.REPORT,
+                    Tables.ORDERLYWEB_REPORT_TAG.TAG)
+                    .from(Tables.ORDERLYWEB_REPORT_TAG)
+                    .where(Tables.ORDERLYWEB_REPORT_TAG.REPORT.`in`(reportNames))
+                    .groupBy { it[Tables.ORDERLYWEB_REPORT_TAG.REPORT] }
+                    .mapValues { it.value.map { r -> r[Tables.ORDERLYWEB_REPORT_TAG.TAG] }.sorted() }
+        }
+    }
+
+    override fun tagReport(reportName: String, tag: String)
```

Comment:
Ah, so these endpoints are posting a single tag string? Maybe it should be the full list since it looks like we're going to go with a modal-type 'save all the changes now' approach - and also now the endpoints are '../tags' it really will be confusing if it only gets one!